include/eve/detail/function/simd/x86/load.hpp

```diff
@@ -11,125 +11,110 @@
 #pragma once
 
 #include <eve/as.hpp>
+#include <eve/concept/memory.hpp>
 #include <eve/concept/vectorizable.hpp>
 #include <eve/detail/abi.hpp>
-#include <eve/detail/meta.hpp>
-#include <eve/memory/aligned_ptr.hpp>
+#include <eve/detail/category.hpp>
 
 namespace eve::detail
 {
   //================================================================================================
-  // 128 bits loads
+  // Regular loads
   //================================================================================================
-  template<real_scalar_value T, typename N>
-  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, eve::x86_128_ const &, T const* p)
+  template<scalar_pointer Ptr, typename T, typename N, x86_abi ABI>
+  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, ABI const &, Ptr p)
+  requires( std::same_as<T, std::remove_cvref_t<decltype(*p)>> )
   {
-    if constexpr( N::value * sizeof(T) == x86_128_::bytes )
+    if constexpr( !std::is_pointer_v<Ptr> )
     {
-      if constexpr( std::is_same_v<T, double> )
-      {
-        return _mm_loadu_pd(p);
-      }
-      else if constexpr( std::is_same_v<T, float> )
-      {
-        return _mm_loadu_ps(p);
-      }
-      else if constexpr( std::is_integral_v<T> )
-      {
-        return _mm_loadu_si128((__m128i *)p);
-      }
+      return load(eve::as_<wide<T, N>>{}, ABI{}, p.get());
     }
     else
     {
-      typename wide<T, N>::storage_type that;
-      std::memcpy(&that, p, N::value * sizeof(T));
-      return that;
-    }
-  }
+      constexpr auto cat = categorize<wide<T, N>>();
 
-  template<real_scalar_value T, typename N, std::size_t A>
-  EVE_FORCEINLINE auto
-  load(eve::as_<wide<T, N>> const &tgt, eve::x86_128_ const &mode, aligned_ptr<T const, A> p) noexcept
-  {
-    if constexpr( A >= 16 && N::value * sizeof(T) == x86_128_::bytes )
-    {
-      if constexpr( std::is_same_v<T, double> )
+            if constexpr( cat == category::float64x4 )  return _mm256_loadu_pd(p);
+      else  if constexpr( cat == category::float64x2 )  return _mm_loadu_pd(p);
+      else  if constexpr( cat == category::float32x8 )  return _mm256_loadu_ps(p);
+      else  if constexpr( cat == category::float32x4 )  return _mm_loadu_ps(p);
+      else  if constexpr( cat && category::integer_ && N::value*sizeof(T) == x86_256_::bytes)
       {
-        return _mm_load_pd(p.get());
+        return _mm256_loadu_si256((__m256i *)p);
       }
-      else if constexpr( std::is_same_v<T, float> )
+      else  if constexpr( cat && category::integer_ && N::value*sizeof(T) == x86_128_::bytes)
       {
-        return _mm_load_ps(p.get());
+        return _mm_loadu_si128((__m128i *)p);
       }
-      else if constexpr( std::is_integral_v<T> )
+      else
       {
-        return _mm_load_si128((__m128i *)p.get());
+        typename wide<T, N>::storage_type that;
+        std::memcpy(&that, p, N::value * sizeof(T));
+        return that;
```

Comment:
Do you really need this `else` in `x86`? I think this should give you a compiler error, which it will if you do nothing.

Reply:
this was meant to be the case for wide smaller than the expected cardinal. But now that I look at it, if you have a small real, it calls loadu. This maybe need to be generalized to other cases no or can I loadu past my data ?